go/libkb/client.go

```diff
@@ -138,7 +139,15 @@ func NewClient(e *Env, config *ClientConfig, needCookie bool) *Client {
 
 	var timeout time.Duration
 
-	xprt := *(http.DefaultTransport.(*http.Transport)) // shallow copying
+	xprt := http.Transport{ // from http.DefaultTransport
+		Proxy: http.ProxyFromEnvironment,
+		Dial: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}).Dial,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
```

Comment:
It looks like by not sharing a transport, we lose cached connections between different `Client` instances. Although this is still a temporary solution. When go1.7 is released (and after we switching to it),  we can revert to the old logic where `Transport` is only [created under certain conditions](https://github.com/keybase/client/commit/2f5ea3745eed0095f5c844482bda79c5c3fc101d#diff-d58cb89c13006e531a58d82658855f64L141).